src/ttsim/tt_dag_elements/param_objects.py

```diff
@@ -97,29 +98,50 @@ class PiecewisePolynomialParam(ParamObject):
 
 
 @dataclass(frozen=True)
-class ConsecutiveInt1dLookupTableParam(ParamObject):
+class ConsecutiveIntLookupTableParam(ParamObject):
     """A parameter with its contents read and converted from a YAML file.
 
-    Its value is a ConsecutiveInt1dLookupTableParamValue object, i.e., it contains the
+    Its value is a ConsecutiveInt2dLookupTableParamValue object, i.e., it contains the
     parameters for calling `lookup_table`.
     """
 
-    value: ConsecutiveInt1dLookupTableParamValue = PLACEHOLDER_FIELD  # type: ignore[assignment]
+    value: ConsecutiveIntLookupTableParamValue = PLACEHOLDER_FIELD  # type: ignore[assignment]
     note: str | None = None
     reference: str | None = None
 
 
-@dataclass(frozen=True)
-class ConsecutiveInt2dLookupTableParam(ParamObject):
-    """A parameter with its contents read and converted from a YAML file.
+class ConsecutiveIntLookupTableParamValue:
+    """The parameters expected by lookup_table"""
 
-    Its value is a ConsecutiveInt2dLookupTableParamValue object, i.e., it contains the
-    parameters for calling `lookup_table`.
-    """
+    bases_to_subtract: Int[Array, "n_rows n_cols"]
+    lookup_multipliers: Int[Array, "n_rows n_cols"]
+    values_to_look_up: Float[Array, "n_rows n_cols"]
+    xnp: ModuleType
+
+    def __init__(
+        self,
+        xnp: ModuleType,
+        values_to_look_up: Float[Array, "n_rows n_cols"],
+        bases_to_subtract: Int[Array, "n_rows n_cols"],
+    ) -> None:
+        self.xnp = xnp
+        self.values_to_look_up = values_to_look_up.flatten()
+        self.bases_to_subtract = xnp.expand_dims(bases_to_subtract, axis=1)
+        self.lookup_multipliers = xnp.concatenate(
+            [
+                (xnp.cumprod(xnp.asarray(values_to_look_up.shape)[::-1])[::-1])[1:],
+                xnp.asarray([1]),
+            ]
+        )
 
-    value: ConsecutiveInt2dLookupTableParamValue = PLACEHOLDER_FIELD  # type: ignore[assignment]
-    note: str | None = None
-    reference: str | None = None
+    def lookup(
+        self: ConsecutiveIntLookupTableParamValue, *args: int
+    ) -> Float[Array, "n_rows n_cols"]:
```

Comment:
I don't quite get the type hint here. I suppose this is still at the level of the individual row (no vectorization to `n_obs`, hence `int` in the arguments)? Then we should also return a scalar `float | int | bool`.

If the annotation is at the column level, we should hav `*args: IntColumn` and return `FloatColumn | IntColumn | BoolColumn`.